### PR TITLE
Yvm exec

### PR DESCRIPTION
**Pull request by brandonbaksh**
- implemented yvm exec command
- usage: `yvm exec <version> <yarn commands>`

## Dev prep
- use yvm install to get 1.7.0 and 1.0.2

## QA Steps
- `make install` and open a new terminal (or `source /usr/local/bin/yvm`)
- use `yvm exec 1.7.0 versions`
- use `yvm exec 1.0.2 versions`
- ensure the correct output from both
- navigate to fe-thm, use `yvm exec 1.0.2 upgrade-interactive` and ensure it works as expected

### CODE REVIEW

File: src/commands/exec.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+
+const getYarnCLI = version => (
```
Comment (francoiscampbell) [inline src/commands/exec.js]: this isn't the CLI file any more, i'd just make a function called `runYarn` that massages the `argv` and calls `require(path.resolve(getYarnPath(version), 'bin/yarn.js'))`

File: src/index.js [hunk 2]
```diff
@@ -16,9 +16,12 @@ module.exports = function(args) {
...
 
     argParser
-        .command('manage <version>')
-        .action(version => {
-            console.log(`Managing yarn v${version}`)
+        .command('exec <version>')
+        .action((version) => {
+            console.log(`Executing yarn command with version ${version}`)
+            const exec = require('./commands/exec')
+            const extraArgs = args.slice(4)  //returns args without [npm, start, exec, <version>, <command>]
```
Comment (francoiscampbell) [inline src/index.js]: seems `commander` can glob the rest if you use `...`: https://www.npmjs.com/package/commander#variadic-arguments

you could probably just do 
```
.command('exec <version> [extrasArgs...]')`
.action((version, extraArgs) => {
```

File: src/index.js [hunk 3]
```diff
@@ -16,9 +16,12 @@ module.exports = function(args) {
         })
 
     argParser
-        .command('manage <version>')
-        .action(version => {
-            console.log(`Managing yarn v${version}`)
+        .command('exec <version>')
+        .action((version) => {
```
Comment (umar-tophat) [inline src/index.js]: I think you can add an extra arg on this line and it will contain the rest of the args to the command.

Comment (umar-tophat) [reply src/index.js]: something like `(version, rest) => {`

**Comment by umar-tophat** [comment]
## ✅ QA

- `npm start exec <version> versions` shows the correct version for each version number
- If I use a version that has not been installed, I get a nice suggestion on how to install
- `upgrade-interactive` works as expected.

File: src/commands/exec.js [hunk 4]
```diff
@@ -1,19 +1,21 @@
+/* eslint-disable global-require,prettier/prettier,import/no-dynamic-require */
```
Comment (francoiscampbell) [inline src/commands/exec.js]: why disable prettier?

Comment (brandonbaksh) [reply src/commands/exec.js]: rip i just copy pasta'd that from a different file

**Pull request merged**